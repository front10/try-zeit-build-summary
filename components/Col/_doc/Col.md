# Col
  [![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
  

  ## Properties
  | </br>Name | </br>Type | </br>Summary | </br>Default | 
| ---- | ---- | ---- | ---- |
| children | `union` | Col size on sm. | `null` |
| lg | `enum` | Col size on lg. | `null` |
| md | `enum` | Col size on md. | `null` |
| sm | `enum` | Col size on sm. | `null` |
| xl | `enum` | Children to show in component | `null` |
| xs | `enum` | Col size on xs. | `null` |