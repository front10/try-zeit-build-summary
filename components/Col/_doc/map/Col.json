{"description":"","displayName":"Col","methods":[],"props":{"className":{"type":{"name":"string"},"required":false,"description":"Class name of span","defaultValue":{"value":"''","computed":false}},"style":{"type":{"name":"objectOf","value":{"name":"any"}},"required":false,"description":"Css style applied to the component","defaultValue":{"value":"null","computed":false}},"xs":{"type":{"name":"enum","value":[{"value":"''","computed":false},{"value":"'1'","computed":false},{"value":"'2'","computed":false},{"value":"'3'","computed":false},{"value":"'4'","computed":false},{"value":"'5'","computed":false},{"value":"'6'","computed":false},{"value":"'7'","computed":false},{"value":"'8'","computed":false},{"value":"'9'","computed":false},{"value":"'10'","computed":false},{"value":"'11'","computed":false},{"value":"'12'","computed":false}]},"required":false,"description":"Col size on xs.","defaultValue":{"value":"null","computed":false}},"sm":{"type":{"name":"enum","value":[{"value":"''","computed":false},{"value":"'1'","computed":false},{"value":"'2'","computed":false},{"value":"'3'","computed":false},{"value":"'4'","computed":false},{"value":"'5'","computed":false},{"value":"'6'","computed":false},{"value":"'7'","computed":false},{"value":"'8'","computed":false},{"value":"'9'","computed":false},{"value":"'10'","computed":false},{"value":"'11'","computed":false},{"value":"'12'","computed":false}]},"required":false,"description":"Col size on sm.","defaultValue":{"value":"null","computed":false}},"md":{"type":{"name":"enum","value":[{"value":"''","computed":false},{"value":"'1'","computed":false},{"value":"'2'","computed":false},{"value":"'3'","computed":false},{"value":"'4'","computed":false},{"value":"'5'","computed":false},{"value":"'6'","computed":false},{"value":"'7'","computed":false},{"value":"'8'","computed":false},{"value":"'9'","computed":false},{"value":"'10'","computed":false},{"value":"'11'","computed":false},{"value":"'12'","computed":false}]},"required":false,"description":"Col size on md.","defaultValue":{"value":"null","computed":false}},"lg":{"type":{"name":"enum","value":[{"value":"''","computed":false},{"value":"'1'","computed":false},{"value":"'2'","computed":false},{"value":"'3'","computed":false},{"value":"'4'","computed":false},{"value":"'5'","computed":false},{"value":"'6'","computed":false},{"value":"'7'","computed":false},{"value":"'8'","computed":false},{"value":"'9'","computed":false},{"value":"'10'","computed":false},{"value":"'11'","computed":false},{"value":"'12'","computed":false}]},"required":false,"description":"Col size on lg.","defaultValue":{"value":"null","computed":false}},"xl":{"type":{"name":"enum","value":[{"value":"''","computed":false},{"value":"'1'","computed":false},{"value":"'2'","computed":false},{"value":"'3'","computed":false},{"value":"'4'","computed":false},{"value":"'5'","computed":false},{"value":"'6'","computed":false},{"value":"'7'","computed":false},{"value":"'8'","computed":false},{"value":"'9'","computed":false},{"value":"'10'","computed":false},{"value":"'11'","computed":false},{"value":"'12'","computed":false}]},"required":false,"description":"Children to show in component","defaultValue":{"value":"null","computed":false}},"children":{"type":{"name":"union","value":[{"name":"arrayOf","value":{"name":"node"}},{"name":"node"}]},"required":false,"description":"Col size on sm.","defaultValue":{"value":"null","computed":false}}},"filename":"/Users/darielnoel/repositories/front10/airmodules/src/components/elements/Col/Col.jsx"}