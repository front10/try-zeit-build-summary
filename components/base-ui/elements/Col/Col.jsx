import React from 'react';
import PropTypes from 'prop-types';
import useStyles from '@front10/helpers/dist/UseStyles';
import BaseBox from '../_commons/BaseBox';

const Col = props => {
  const { xs, sm, md, lg, xl } = props;
  const [className, style] = useStyles(props);

  let cls = '';
  if (xs) cls = `fr-col-${xs}`;
  else if (!sm && !md && !lg && !xl) cls = 'fr-col';
  if (sm) cls += ` fr-col-sm-${sm}`;
  if (md) cls += ` fr-col-md-${md}`;
  if (lg) cls += ` fr-col-lg-${lg}`;
  if (xl) cls += ` fr-col-xl-${xl}`;
  const newClassName = `${cls} ${className} `;

  return <BaseBox {...props} className={newClassName} style={style} />;
};

Col.propTypes = {
  /**
   * Col size on xs.
   */
  xs: PropTypes.oneOf(['', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
  /**
   * Col size on sm.
   */
  sm: PropTypes.oneOf(['', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
  /**
   * Col size on md.
   */
  md: PropTypes.oneOf(['', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
  /**
   * Col size on lg.
   */
  lg: PropTypes.oneOf(['', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
  /**
   * Children to show in component
   */
  xl: PropTypes.oneOf(['', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
  /**
   * Col size on sm.
   */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};
Col.defaultProps = {
  xs: null,
  sm: null,
  md: null,
  lg: null,
  xl: null,
  children: null
};

Col.componentName = 'Col';

export default Col;
