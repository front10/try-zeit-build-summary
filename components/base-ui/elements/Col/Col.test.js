import React from 'react';
import { render } from 'test-utils';
import Col from './Col';


test('render all', () => {
  const { container } = render(<Col />);
  expect(container).toMatchSnapshot();
});
