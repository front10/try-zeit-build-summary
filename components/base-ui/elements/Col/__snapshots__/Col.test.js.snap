// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all 1`] = `
<div
  className="fr-col"
  data-testid={null}
  style={
    Object {
      "filter": undefined,
    }
  }
/>
`;
